### PR TITLE
Add loading spinner to contract interaction's ABI fetching and parsing

### LINKED ISSUE #3804: Add Loading Spinner to contract interaction's ABI fetching and parsing
### Story

As a user, when selecting a contract to interact with, I want to see a loading spinner as long as the ABI of the contract is fetched and parsed.

### Description

When you select a contract address in the contract interaction section, a spinner should appear so long as the ABI is fetched and parsed, so that the user has an indication that the process is running instead of nothing happening for some 1 second or a few (at worst). This spinner should appear where the ABI text field and function selector are.

### This issue is complete when...

- [ ] A spinner appears if a contract address is selected. (Or in other words, every time a contract ABI is fetched and parsed for useful methods).
- [ ] The spinner disappears if the useful methods are extracted and made available for the user's selection.

### PR DESCRIPTION
**Pull request by Julianahlk**
## Description

This PR adds a loading spinner to the contract interaction ABI. The spinner should appear as long as the ABI is being fetched and parsed. It should appear for at least 1-5 seconds. 

**For testing:**
- It would be easier to slow down your network speed as the spinner only shows up 1-2 secs.

network

**Changes** 🏗
- Added `SpinnerLoader`

isloadingabi

---
Resolves #3804

### CODE REVIEW

**Review by jakubcolony — APPROVED** [review]
Looks great! I was about to mention the misalignment but you pushed the fix first 😁! 

image

Left you one comment about `console.log`:

File: src/modules/dashboard/components/Dialogs/ControlSafeDialog/TransactionTypesSection/ContractInteractionSection.tsx [hunk 1]
```diff
@@ -234,6 +239,8 @@ const ContractInteractionSection = ({
     transactionFormIndex,
   ]);
 
+  // console.log('loadingABI', loadingABI);
```
[the remark this answers is not in the document]
Comment (Julianahlk) [reply src/modules/dashboard/components/Dialogs/ControlSafeDialog/TransactionTypesSection/ContractInteractionSection.tsx:242]: Thanks for catching this @jakubcolony 😉

**Review by willm30 — COMMENTED** [review]
Would it be possible to style like the Transfer NFT one for consistency?

nft

**Review by danbr — APPROVED** [review]
LGTM - well done....

**Review by chinins — APPROVED** [review]
Good to go!

File: src/modules/dashboard/components/Dialogs/ControlSafeDialog/TransactionTypesSection/ContractInteractionSection.tsx [hunk 2]
```diff
@@ -17,32 +17,37 @@ import {
...
+    id: `dashboard.ControlSafeDialog.TransactionTypesSection.ContractInteractionSection.userPickerPlaceholder`,
     defaultMessage: 'Select or paste a contract address',
   },
+  loadingContract: {
+    id: `dashboard.ControlSafeDialog.TransactionTypesSection.ContractInteractionSection.loadingContract`,
+    defaultMessage: 'Loading Contract',
+  },
 });
 
 const displayName = `dashboard.ControlSafeDialog.ControlSafeForm.ContractInteractionSection`;
```
Comment (willm30) [inline src/modules/dashboard/components/Dialogs/ControlSafeDialog/TransactionTypesSection/ContractInteractionSection.tsx:53]: Could change this also if you're updating the others

**Review by willm30 — APPROVED** [review]
LGTM

**Pull request merged**